skimage/io/_plugins/matplotlib_plugin.py

```diff
@@ -1,12 +1,154 @@
+
+        - signed: whether the image has negative values.
+        - out_of_range_float: if the image has floating point data
+          outside of [-1, 1].
+        - low_dynamic_range: if the image is in the standard image
+          range (e.g. [0, 1] for a floating point image) but its
+          dynamic range would be too small to display with standard
+          image ranges.
+        - unsupported_dtype: if the image data type is not a
+          standard skimage type, e.g. ``numpy.uint64``.
+    """
+    immin, immax = np.min(image), np.max(image)
+    imtype = image.dtype.type
+    try:
+        lo, hi = dtypes.dtype_range[imtype]
+    except KeyError:
+        lo, hi = immin, immax
+
+    signed = immin < 0
+    out_of_range_float = (np.issubdtype(image.dtype, np.float) and
+                          (immin < lo or immax > hi))
+    low_dynamic_range = (immin != immax and
+                         (float(immax - immin) / (hi - lo)) < (1. / 255))
+    unsupported_dtype = image.dtype not in dtypes._supported_types
+
+    ip = ImageProperties(signed, out_of_range_float,
+                         low_dynamic_range, unsupported_dtype)
+    return ip
+
+
+def _raise_warnings(image_properties):
+    """Raise the appropriate warning for each nonstandard image type.
+
+    Parameters
+    ----------
+    image_properties : ImageProperties named tuple
+        The properties of the considered image.
+    """
+    ip = image_properties
+    if ip.unsupported_dtype:
+        warnings.warn("Non-standard image type; displaying image with "
+                      "stretched contrast.")
+    if ip.low_dynamic_range:
+        warnings.warn("Low image dynamic range; displaying image with "
+                      "stretched contrast.")
+    if ip.out_of_range_float:
+        warnings.warn("Float image out of standard range; displaying image "
+                      "with stretched contrast.")
+
+
+def _get_display_range(image):
+    """Return the display range for a given set of image properties.
+
+    Parameters
+    ----------
+    image : array
+        The input image.
+
+    Returns
+    -------
+    lo, hi : same type as immin, immax
+        The display range to be used for the input image.
+    cmap : string
+        The name of the colormap to use.
+    """
+    ip = _get_image_properties(image)
+    immin, immax = np.min(image), np.max(image)
+    if ip.signed:
+        magnitude = max(abs(immin), abs(immax))
+        lo, hi = -magnitude, magnitude
+        cmap = _diverging_colormap
+    elif any(ip):
+        _raise_warnings(ip)
+        lo, hi = immin, immax
+        cmap = _nonstandard_colormap
+    else:
+        lo = 0
+        imtype = image.dtype.type
+        hi = dtypes.dtype_range[imtype][1]
+        cmap = _default_colormap
+    return lo, hi, cmap
+
+
+def imshow(im, *args, **kwargs):
+    """Show the input image and return the current axes.
+
+    By default, the image is displayed in greyscale, rather than
```

Comment:
Why not just say instead of the default colormap and leave it at that.  It will be trie for all versions of mpl.